### PR TITLE
Cleanup Web3DOverlay and tablet home button

### PR DESCRIPTION
**Pull request by SamGondelman**
- Remove extraneous "resolution" field from web overlays.  This fixes a culling issue and makes them more closely match web entities.  Before, web overlays would render with the wrong dimensions.
- Copy intersection logic from web entities to web overlays to get rid of a TODO.
- Update tablet scripts to account for overlay changes.
- Fix positioning of tablet screen, home button, and highlight at different avatar scales ([FB9035](https://highfidelity.fogbugz.com/f/cases/9035) and probably [FB8906](https://highfidelity.fogbugz.com/f/cases/8906)).
- Replace the home button overlay with a circle overlay ([FB8214](https://highfidelity.fogbugz.com/f/cases/8214)).
- Fix reticle depth at different avatar scales.
- Fix tablet "x" button triggering on mouse click.

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/aef57ea4aa6304a3149aab964d4865dc/raw/7b50ff9333f65a9f27e5bb9070b4e7610bbe3a02/WebOverlayTest.js).  It'll make a web overlay and a web entity in front of you.  Verify that they are the same size.
- Press '1' and '2' to increase and decrease their "dpi" settings.  Their dimensions should not change.
- Enable tablet in desktop mode (Developer -> UI).  You should be able to mouse over the tablet and the home button and have them highlight as expected.
- Change avatar scales.  The tablet screen and highlight effect should resize properly and not be offset from the expected positions.  Try a bunch of different scales.
- In HMD, lasers, mouse, and stylus should all work on the tablet and home button.  The highlight should be in the right place.  Try at different avatar scales.
- Everything (interacting with tablet, hovering over home button, etc) should still work when you grab the tablet with one hand and use the other hand to interact with it.
- If you look at the tablet from the side and laser onto the home button, the laser should be hitting very close to the surface of the tablet.  There should not be a noticeable space in between.
- The mouse depth and size should be correct at different avatar scales.
- With the mouse, the "x" button should only trigger on mouse release, not mouse press.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5055/HighFidelity-Beta-PR11847-90c4af7f41565ac694624ff6917ef49953751261.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5055/HighFidelity-Beta-PR11847-90c4af7f41565ac694624ff6917ef49953751261.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5057/HighFidelity-Beta-PR11847-393ad0079c4af2f09048aa5e18013129d4ba38de.exe

**Comment by SamGondelman** [comment]
build this please you devil bot

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5068/HighFidelity-Beta-PR11847-393ad0079c4af2f09048aa5e18013129d4ba38de.dmg

**Comment by SamGondelman** [comment]
build this please

p l e a s e

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5070/HighFidelity-Beta-PR11847-393ad0079c4af2f09048aa5e18013129d4ba38de.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5070/HighFidelity-Beta-PR11847-393ad0079c4af2f09048aa5e18013129d4ba38de.dmg

**Comment by marko8904** [comment]
ubuntu build failed unexpectedly. build this please.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5077/HighFidelity-Beta-PR11847-393ad0079c4af2f09048aa5e18013129d4ba38de.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5077/HighFidelity-Beta-PR11847-393ad0079c4af2f09048aa5e18013129d4ba38de.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5180/HighFidelity-Beta-PR11847-9733de62977aceb14a0e091dda1d3552401bcda8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5185/HighFidelity-Beta-PR11847-52039e022b197f8a87a10ce84871e526450bc67f.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5187/HighFidelity-Beta-PR11847-52039e022b197f8a87a10ce84871e526450bc67f.exe

**Comment by SamGondelman** [comment]
adsfiojaosdifjasdoifj build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5188/HighFidelity-Beta-PR11847-52039e022b197f8a87a10ce84871e526450bc67f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5188/HighFidelity-Beta-PR11847-52039e022b197f8a87a10ce84871e526450bc67f.dmg

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

Run this. It'll make a web overlay and a web entity in front of you. Verify that they are the same size.
**Check**
Press '1' and '2' to increase and decrease their "dpi" settings. Their dimensions should not change.
**Check, see images for examples of higher and lower DPI working with the web overlay & entity**
image

image


Enable tablet in desktop mode (Settings -> General). You should be able to mouse over the tablet and the home button and have them highlight as expected.
**Maybe I'm missing something but I do not see this option anymore in window described? - but when in HMD highlighting works with mouse as expected. Unsure if this is a fail?**

Change avatar scales. The tablet screen and highlight effect should resize properly and not be offset from the expected positions. Try a bunch of different scales.
**Pass**

In HMD, lasers, mouse, and stylus should all work on the tablet and home button. The highlight should be in the right place. Try at different avatar scales.
**Fail, when very small (small as you can get in a sandbox domain with default settings) it appears the mouse is offset to where it is actually pointing for the desired highlight of tablet icons**

Everything (interacting with tablet, hovering over home button, etc) should still work when you grab the tablet with one hand and use the other hand to interact with it.
**Fail, This may be a small issue but clicking the X of tablet seems to "double press" for example if I'm in Create and press the X on tablet, it doesn't take me to menu, it double-taps and closes. If I press and hold, but move mouse off the X then it seems to do 1 click as desired, so I imagine it's registering 2 upon the press of mouse button and the letting go of mouse button.**

If you look at the tablet from the side and laser onto the home button, the laser should be hitting very close to the surface of the tablet. There should not be a noticeable space in between.
**Pass**


A few small issues noted, **Fail**

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5190/HighFidelity-Beta-PR11847-d4a4c8902ae069f97fe2f5baafe8cfa733d1929b.dmg

**Comment by SamGondelman** [comment]
build this please you piece of GARBAGE

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5191/HighFidelity-Beta-PR11847-d4a4c8902ae069f97fe2f5baafe8cfa733d1929b.dmg

**Comment by marko8904** [comment]
build this please

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5197/HighFidelity-Beta-PR11847-d4a4c8902ae069f97fe2f5baafe8cfa733d1929b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5197/HighFidelity-Beta-PR11847-d4a4c8902ae069f97fe2f5baafe8cfa733d1929b.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

Run this. It'll make a web overlay and a web entity in front of you. Verify that they are the same size.
**Check**
Press '1' and '2' to increase and decrease their "dpi" settings. Their dimensions should not change.
**Check**
Enable tablet in desktop mode (Developer -> UI). You should be able to mouse over the tablet and the home button and have them highlight as expected.
**Check**
Change avatar scales. The tablet screen and highlight effect should resize properly and not be offset from the expected positions. Try a bunch of different scales.
**Check**
In HMD, lasers, mouse, and stylus should all work on the tablet and home button. The highlight should be in the right place. Try at different avatar scales.
**Check**
Everything (interacting with tablet, hovering over home button, etc) should still work when you grab the tablet with one hand and use the other hand to interact with it.
**Check**
If you look at the tablet from the side and laser onto the home button, the laser should be hitting very close to the surface of the tablet. There should not be a noticeable space in between.
**Check**
The mouse depth and size should be correct at different avatar scales.
**Check**
With the mouse, the "x" button should only trigger on mouse release, not mouse press.
**Check**

All solid **Pass**

File: scripts/system/libraries/WebTablet.js [hunk 1]
```diff
@@ -118,15 +118,16 @@ WebTablet = function (url, width, dpi, hand, clientOnly, location, visible) {
         Overlays.deleteOverlay(this.webOverlayID);
     }
 
-    var WEB_ENTITY_Z_OFFSET = (tabletDepth / 2) * (1 / sensorScaleFactor);
-    var WEB_ENTITY_Y_OFFSET = 0.004 * (1 / sensorScaleFactor);
-
+    var WEB_ENTITY_Z_OFFSET = (tabletDepth / 2.0) / sensorScaleFactor;
+    var WEB_ENTITY_Y_OFFSET = 0.004;
+    var screenWidth = 0.82 * tabletWidth;
+    var screenHeight = 0.81 * tabletHeight;
```
Comment (jherico) [inline scripts/system/libraries/WebTablet.js]: why are these different?

Comment (SamGondelman) [reply scripts/system/libraries/WebTablet.js]: The screen doesn't take up an equal portion of the tablet model left/right vs. up/down.

File: interface/src/ui/overlays/Web3DOverlay.h [hunk 2]
```diff
@@ -93,8 +91,7 @@ public slots:
     gpu::TexturePointer _texture;
     QString _url;
     QString _scriptURL;
-    float _dpi;
-    vec2 _resolution{ 640, 480 };
+    float _dpi { 30 };
```
Comment (NissimHadar) [inline interface/src/ui/overlays/Web3DOverlay.h]: `30` -> `30.0f`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5460/HighFidelity-Beta-PR11847-99c52f86021fc1f6a919d8aea6de53d8633f56a8.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5466/HighFidelity-Beta-PR11847-99c52f86021fc1f6a919d8aea6de53d8633f56a8.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5481/HighFidelity-Beta-PR11847-99c52f86021fc1f6a919d8aea6de53d8633f56a8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5481/HighFidelity-Beta-PR11847-99c52f86021fc1f6a919d8aea6de53d8633f56a8.dmg

**Pull request merged**